cryptarchia/cryptarchia.py

```diff
@@ -173,28 +173,38 @@ def nullifier(self) -> Id:
 
 
 @dataclass
-class MockLeaderProof:
-    commitment: Id
+class LeaderProof:
     nullifier: Id
     evolved_commitment: Id
     slot: Slot
-    parent: Id
+    # The produced proof
 
     @staticmethod
-    def new(coin: Coin, slot: Slot, parent: Id):
+    def new(#Public inputs
+            slot: Slot,
+            _epoch_nonce: Id,
+            _threshold_0: int,
+            _threshold_1: int,
+            _eligible_notes: set[Id],
+
+            #Secret inputs
+            coin: Coin,
+            #membership witness
+            ):
         evolved_coin = coin.evolve()
 
-        return MockLeaderProof(
-            commitment=coin.commitment(),
+        return LeaderProof( # TODO: generate the proof with risc0
             nullifier=coin.nullifier(),
             evolved_commitment=evolved_coin.commitment(),
             slot=slot,
-            parent=parent,
         )
 
-    def verify(self, slot: Slot, parent: Id):
-        # TODO: verification not implemented
-        return slot == self.slot and parent == self.parent
+    def verify(self, slot: Slot,
+               _epoch_nonce: Id,
+               _threshold_0: int,
+               _threshold_1: int,
+               _eligible_notes: set[Id],):
+        return slot == self.slot #TODO: and verification of the proof with risc0
```

Comment:
We can write out the verification constraints in python. I.e. this serves as a spec for the NIZK statements.